controllers/config_helper.go

```diff
@@ -203,14 +169,18 @@ func (r *GatekeeperReconciler) handleConfigController(ctx context.Context) error
 	}
 
 	r.isConfigCtrlRunning = true
+	r.subControllerWait.Add(1)
 
 	// Use another go routine for the Config controller
 	go func() {
-		err := configMgr.Start(ctx)
+		err := configMgr.Start(configCtrlCtx)
 		if err != nil {
 			setupLog.Error(err, "A problem running Config manager. Triggering a reconcile to restart it.")
 		}
 
+		defer r.configCtrlCtxCancel()
+
+		r.configCtrlCtxCancel = nil
```

Comment:
I'd rather not set this to `nil` to avoid a potential race conditions where another goroutine tries to cancel the context with this set to `nil` and causing a panic. If the context is already canceled, then if the cancel func is called again, is just a no op.